src/basic/generate-af-list.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+cpp -dM -include sys/socket.h - </dev/null | \
```

Comment:
Yes, you have to use a compiler here, and what you want to do is to pass the compiler array to the script as arguments and run it. Something like this:

```meson
cc_array = cc.cmd_array()
generate_af_list = find_program('generate-af-list.sh')
af_list_txt = custom_target(
    'af-list.txt',
    output : 'af-list.txt',
    command : [generate_af_list] + cmd_array,
    capture : true)
```

Then inside the shell script, have:

```sh
"$@" -E -dM -include sys/socket.h - </dev/null | \
    grep -Ev 'AF_UNSPEC|AF_MAX' | \
    awk '/^#define[ \t]+AF_[^ \t]+[ \t]+PF_[^ \t]/ { print $2; }'
```

If any of your scripts take arguments, you can use `shift`, and so on as usual.